### ISSUE TITLE
[Regression] Flickering Highlights on Phantasy Star Portable 2 Infinity

### ISSUE THREAD

**Issue by Sazonde99** [hrydgard/ppsspp#16858]
### Game or games this happens in

NPJH50332 - Phantasy Star Portable 2 Infinity

### What area of the game

Sometimes when you load into the game, there's a pink (sometimes rainbow) line at the edge of the screen that flickers. Most of the time its at the upper right corner but sometimes the lower edge as well. The issue can happen when I load into the City or a Mission. Even loading screens can be affected. About a 50/50 chance for it to occur whenever I load into the game into an area. I tried each of the backends and it happens on all of them.

Not entirely sure if there's any relation but the axe's beam blade also jitters.

Changing the render resolution may shift where the line is. Changing it back and forth to the original render resolution can make the effect worse and increase the amount of surface area.

Attached is a savestate from 1.13.2 from which the following videos and image derive from. I tested the savestate in 1.14+ and attempted to reproduce directly from a new save to confirm.

[NPJH50332_1.02_0.zip](https://github.com/hrydgard/ppsspp/files/10528534/NPJH50332_1.02_0.zip)

### What happens

https://user-images.githubusercontent.com/123838641/215296202-9d2e32ae-4533-4851-9fd2-a27037ebf82d.mp4

![image](https://user-images.githubusercontent.com/123838641/215295227-5baf2292-fda2-4699-ba25-f8fd333c60ce.png)

(Bottom and Right Side of the Image)

### What should happen

https://user-images.githubusercontent.com/123838641/215295360-a64ef18a-f2ff-4b40-8582-38c5fb8ba0c9.mp4

1.13.2 is the last version where the flickering highlights do not happen. Any version from 1.14 and up has this issue including the bleeding edge builds.

The glowing axe blade also does not jitter.

### Platform

Windows

### Mobile phone model or graphics card

NVIDIA TITAN Xp

### PPSSPP version affected

v1.14

### Last working version

v1.13.2

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [X] Include logs or screenshots of issue.

**Comment by sum2012** [comment]
How does real PSP look like ?

**Comment by hrydgard** [comment]
This doesn't look like something that would happen on the real hardware.

Just curious, does it also happen at 1x rendering?

Either way, we can probably fix or work around this somehow.

**Comment by Sazonde99** [comment]
Indeed, that shouldn't be happening on hardware since its pretty distracting haha. It does not happen on 1x rendering. It happens from 2x and up though as the resolution increases it gets harder to notice due to the amount of pixels it affects shrinks in surface area. Switching between 2x-4x is most noticeable. Can also take a few times of switching back and forth between different render resolutions to make it happen on them.

So I tested this some more in town. Turns out this can be triggered just through changing render resolution many times without needing it to already be on the screen.

![2x](https://user-images.githubusercontent.com/123838641/215356330-43ea5206-c7ad-4569-810b-c14c1a35225a.png)

(Upper Right Edge)

**Comment by hrydgard** [comment]
Almost certainly the cause is that the game does something that makes us detect a 512x272 framebuffer, while the game only renders to the first 480x272 (which is what gets displayed, due to the PSP's resolution). Somehow, some bright garbage pixels end up in the remaining 32x272 to the right, and due to us using bilinear filtering to blit to the final framebuffer (or when inputing into a post-processing shader), a column of pixels of that garbage gets to influence the final results.

Can see some different solutions:

 * See if we can avoid the game detecting the framebuffer as 512x480
 * See if there's a way to avoid garbage data from getting there
 * Change post-processing / final blit to clamp to 480x272

I kinda like the last one best since it's more generic, although it'd require modifications of post-shaders, maybe we can inject some clamping automatically when sampling the texture..

**Comment by Rikitikitavi88** [comment]
This glitch still here. At first it wasn't there, but after a few changes to the graphic settings, it appeared
